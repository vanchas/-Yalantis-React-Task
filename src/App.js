import React, { useState, useEffect } from 'react'
import './App.css'
import MonthsList from './components/months-list/MonthsList'
import UsersList from './components/users-list/UsersList'

function App() {
  const [users, setUsers] = useState(null)
  const [usersToShow, setUsersToShow] = useState(null)

  const fetchUsers = async () => {
    return await fetch('https://yalantis-react-school-api.yalantis.com/api/task0/users')
      .then(res => res.json())
      .then(data => setUsers(data))
      .catch(err => console.error(err))
  }

  useEffect(() => {
    fetchUsers()
  }, [])

  return (
    <div className={`App`}>
      <MonthsList
        users={users}
        setUsersToShow={setUsersToShow}
      />
      <UsersList
        users={usersToShow}
      />
    </div>
  );
}

export default App
