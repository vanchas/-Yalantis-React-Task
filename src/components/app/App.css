.App {
	display: grid;
	grid-template-columns: 1fr 2fr;
	font-family: Arial, Helvetica, sans-serif;
	max-width: 1000px;
	margin: 0 auto;
}
