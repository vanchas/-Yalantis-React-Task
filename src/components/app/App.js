import React from 'react'
import './App.css'
import { Provider } from 'react-redux'
import store from '../../redux/store'
import MonthsList from '../months-list/MonthsList'
import UsersList from '../users-list/UsersList'

function App() {
  return (
    <Provider store={store}>
      <div className={`App`}>
        <MonthsList />
        <UsersList />
      </div>
    </Provider>
  );
}

export default App
