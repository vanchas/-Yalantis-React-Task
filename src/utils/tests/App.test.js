import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import App from '../../App'


describe('Link', () => {
  it('should render correctly', () => {
    const output = shallow(<App />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
})
