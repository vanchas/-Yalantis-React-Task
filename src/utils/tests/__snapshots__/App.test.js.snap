// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<div
  className="App"
>
  <MonthsList
    setUsersToShow={[Function]}
    users={null}
  />
  <UsersList
    users={null}
  />
</div>
`;
